framework/aster-frame/src/task/processor.rs

```diff
@@ -117,9 +119,32 @@ fn switch_to_task(next_task: Arc<Task>) {
 
     let next_task_cx_ptr = &next_task.inner_ctx() as *const TaskContext;
 
+    // If the current and next tasks meet the following conditions,
+    // then we can safely skip switching page table for improved performance.
+    //
+    // Current task         | Next task                 | Should switch page table?
+    // ----------------------------------------------------------------------------
+    // any                  | w/o a user space          | No
+    // w/ a user space.     | w/ the same user space    | No
+    //
+    // In the table above, w/ is short for with and w/o is for without.
+    let vm_to_switch = next_task
```

Comment:
Rename `vm_to_switch` to `next_vm_space` as the latter is more consistent with other variable names (like `next_task` and `vm_space.activate`)